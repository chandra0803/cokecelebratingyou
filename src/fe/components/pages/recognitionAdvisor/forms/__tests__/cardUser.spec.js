/*global jest*/

jest.mock( 'react-redux' );
import React from 'react';
import CardUser from '../cardUser.js';
import renderer from 'react-test-renderer';

window.recognitionAdvisor = {
  raUrl: 'ra/reminders.action',
  raTilePageDisplay: 'yes',
  raDetailPageDisplay: 'no',
  raEndModelPageDisplay: 'no',
  content: []
};

const defaultProps = {
    recognitionAdvisor: {
      raUrl: 'ra/reminders.action',
      raTilePageDisplay: 'yes',
      raDetailPageDisplay: 'no',
      raEndModelPageDisplay: 'no',
      content: {
        'recognition.content.model.info': 'Recognition Advisor',
        'recognition.content.model.info.ra_days_over_due': 'days overdue!',
        'recognition.content.model.info.ra_new_employee': 'New',
        'recognition.content.model.info.ra_recognize': 'Recognize',
        'recognition.content.model.info.ra_caught': 'You all caught up!'
      }
    }
};

const cardUser = [ {
	'id': 92085,
	'firstName': 'Erica',
	'lastName': 'Jalecia Edwards',
	'participantGroupType': 2,
	'avatarUrl': null,
	'positionType': 'Director',
	'otherMgrApprovedDate': null,
	'currentMgrApprovedDate': '01/08/2018',
	'otherMgrLastRecogDays': null,
  'currentMgrLastRecogDays': 225,
	'otherMgrAwardPoints': null,
	'currentMgrAwardPoints': 10,
	'claimIdByCurrentMgr': '10354',
	'claimIdByOtherMgr': null,
	'noOfDaysPastDueDate': null,
	'participantRecDaysColour': 'red',
	'participantRecDaysColourForOthers': null,
	'claimUrlByCurrentMgr': 'https://g5betaqa.performnet.com/g5beta/login.do',
	'claimUrlByOtherMgr': null,
	'countryId': 6037,
	'nodeId': [ '7202' ],
	'recApprovalStatusType': null
} ];

it( 'renders the dom like the snapshot when the props have default values.', () => {
    const content = {
      content: {
        'recognition.content.model.info': 'Recognition Advisor',
        'recognition.content.model.info.ra_days_over_due': 'days overdue!',
        'recognition.content.model.info.ra_new_employee': 'New',
        'recognition.content.model.info.ra_recognize': 'Recognize',
        'recognition.content.model.info.ra_caught': 'You all caught up!'
      }
    };
    const props = {
        ...defaultProps,
        ...content,
        recognitionAdvisorFunc: jest.fn(),
        raEligibleProgramsFunc: jest.fn()
    };

    const groupType = 0;

    const component = renderer.create(
        <div>
            <CardUser cardUser={ cardUser } groupType={ groupType } { ...props } />
        </div>
    );

    const tree = component.toJSON();
    expect( tree ).toMatchSnapshot();
} );
