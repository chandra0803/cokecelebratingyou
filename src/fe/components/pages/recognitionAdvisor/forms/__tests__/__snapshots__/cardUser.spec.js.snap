// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the dom like the snapshot when the props have default values. 1`] = `
<div>
  <div
    className="card card-large highlightedUser"
  >
    <div
      className="card-front"
    >
      <div
        className="recognitionAdvisorStatus"
      >
        <span
          className="icon-clock removeCircle"
        />
        225
         
        days overdue!
      </div>
      <div
        className="card-top"
      >
        <span
          className="avatar"
        >
          <span
            className="avatarContainer"
          >
            <span
              className="avatar-initials"
            >
              E
               
              J
            </span>
          </span>
        </span>
      </div>
      <div
        className="card-details"
      >
        <div
          className="card-details-inner-wrap"
        >
          <div
            className="participant-name"
          >
            <a
              className="itemName"
              data-participant-ids="[ 92085 ]"
              data-title-page-view="Public Profile"
            >
              Erica
               
              Jalecia Edwards
            </a>
          </div>
          <div
            className="participant-info"
          >
            <span
              className="pi-title"
            >
              Director
            </span>
          </div>
        </div>
      </div>
      <div
        className="card-action"
      >
        <button
          className="btnRecognize btn btn-primary btn-block flipButton"
          data-id={92085}
          onClick={[Function]}
          type="button"
        >
          Recognize
        </button>
      </div>
    </div>
  </div>
</div>
`;
